src/main/java/com/faforever/client/vault/replay/WatchButtonController.java

```diff
@@ -1,81 +1,132 @@
 
-  public Button watchButton;
+  public WatchLiveReplaySplitMenuButton watchButton;
+  public MenuItem notifyMeItem;
+  public MenuItem runReplayItem;
   private GameBean game;
   private final I18n i18n;
   private Timeline delayTimeline;
 
-  public WatchButtonController(ReplayService replayService, ClientProperties clientProperties, TimeService timeService, I18n i18n) {
-    this.replayService = replayService;
-    this.clientProperties = clientProperties;
-    this.timeService = timeService;
-    this.i18n = i18n;
-  }
+  private ChangeListener<Pair<Integer, LiveReplayAction>> trackingReplayPropertyListener;
 
   public void initialize() {
     delayTimeline = new Timeline(
         new KeyFrame(Duration.ZERO, event -> onFinished()),
         new KeyFrame(Duration.seconds(1))
     );
     delayTimeline.setCycleCount(Timeline.INDEFINITE);
-
-    watchButton.setDisable(true);
-    watchButton.setOnAction(event -> replayService.runLiveReplay(game.getId()));
+    watchButton.setIsCannotWatchSupplier(() -> !liveReplayService.canWatchReplay(game));
+    initializeMenuItemsByDefault();
   }
 
   public void setGame(GameBean game) {
-    this.game = game;
     Assert.notNull(game, "Game must not be null");
     Assert.notNull(game.getStartTime(), "The game's start must not be null: " + game);
-    if (canWatch()) {
+
+    this.game = game;
+    if (liveReplayService.canWatchReplay(game)) {
       allowWatch();
     } else {
+      initializeListeners();
       updateWatchButtonTimer();
       delayTimeline.play();
     }
   }
 
-  private boolean canWatch() {
-    java.time.Duration duration = getWatchDelayTime();
-    return duration.isZero() || duration.isNegative();
+  private void initializeListeners() {
+    trackingReplayPropertyListener = (observable, oldValue, newValue) ->
+        JavaFxUtil.runLater(() -> {
+          if (newValue != null) {
+            Integer ID = newValue.getKey();
+            LiveReplayAction action = newValue.getValue();
+            if (game.getId().equals(ID)) {
+              switch (action) {
+                case NOTIFY_ME -> {
+                  notifyMeItem.setText(i18n.get("vault.liveReplays.contextMenu.notifyMe.cancel"));
+                  notifyMeItem.setOnAction(event -> liveReplayService.stopTrackingReplay());
+                }
+                case RUN -> {
+                  runReplayItem.setText(i18n.get("vault.liveReplays.contextMenu.runImmediately.cancel"));
+                  runReplayItem.setOnAction(event -> liveReplayService.stopTrackingReplay());
+                }
+              }
+
+              watchButton.pseudoClassStateChanged(TRACKABLE_PSEUDO_CLASS, true);
+              return;
+            }
+          }
+
+          initializeMenuItemsByDefault();
+          watchButton.pseudoClassStateChanged(TRACKABLE_PSEUDO_CLASS, false);
+        });
+    JavaFxUtil.addAndTriggerListener(liveReplayService.getTrackingReplayProperty(), trackingReplayPropertyListener);
```

Comment:
should be a weak listener, otherwise they will never be collected when the controller is destroyed